src/app/tabs/tabs.component.ts

```diff
@@ -38,23 +39,26 @@ export class TabsComponent extends RouterEnter {
     super(router);
     this.newRelic.setPageViewName('tab');
 
-    const role = this.storage.getUser().role;
-    this.utils.getEvent('notification').subscribe(event => {
-      this.noOfTodoItems++;
-    });
-    this.utils.getEvent('event-reminder').subscribe(event => {
-      this.noOfTodoItems++;
-    });
-    this.utils.getEvent('chat:new-message').subscribe(event => {
-      this.tabsService.getNoOfChats().subscribe(noOfChats => {
-        this.noOfChats = noOfChats;
+    this.restrictedAccess = storage.get('singlePageAccess');
+
+    if (this.restrictedAccess === false) {
```

Comment:
if we haven't set `singlePageAccess` in the storage, it will return `null` I think?

Reply:
I suggest we create a function in storage.service `getRestrictedAccess()` like for other keys, so that we make sure it returns false if not set

Reply:
yes, i thought of that too, but i just feeling something isn't right, because we have too much similar purpose purpose function with different name.
- getUser
- setUser
- getReferrer
- setReferrer
- getConfig
- setConfig
- getBooked....ids
- setBooked....ids
- getCountry
- setCountry
- setCurrentChatChannel
- getCurrentChatChannel

and all of them are using the "get" & "set" functions in the almost same way.

Reply:
i can follow the same pattern, but it actually worsen the maintainability & it contributes to the tech debt?